app/src/main/java/org/ometa/lovemonster/service/ResponseParser.java

```diff
@@ -0,0 +1,210 @@
+     *
+     * @param loveJson
+     *      the json object to parse
+     * @return
+     *      the parsed {@link Love}, or null if it cannot be parsed
+     */
+    private Love parseLove(@Nullable final JSONObject loveJson) {
+        if (loveJson == null) {
+            return null;
+        }
+
+        final String reason = loveJson.optString("reason", null);
+        if (reason == null) {
+            return null;
+        }
+
+        final User lover = parseUser(loveJson.optJSONObject("user_from"));
+        if (lover == null) {
+            return null;
+        }
+
+        final User lovee = parseUser(loveJson.optJSONObject("user_to"));
+        if (lovee == null) {
+            return null;
+        }
+
+        final Calendar createdAt = parseDatetime(loveJson.optString("created_at"));
+        if (createdAt == null) {
+            return null;
+        }
+
+        final Love love = new Love(reason, lover, lovee, createdAt);
+
+        love.message = loveJson.optString("message", null);
+        love.isPrivate = loveJson.optBoolean("private_message", false);
+
+        return love;
+    }
+
+    /**
+     * Parses a {@link User} object from the passed json. If any of the required fields cannot be parsed,
+     * this method will return null. If any non-required fields cannot be parsed, then that field will
+     * be set to null. Null-safe (will return null).
+     *
+     * @param userJson
+     *      the json object to parse
+     * @return
+     *      the parsed {@link User}, or null if it cannot be parsed
+     */
+    private User parseUser(@Nullable final JSONObject userJson) {
+        if (userJson == null) {
+            return null;
+        }
+
+        final String email = userJson.optString("email", null);
+        if (email == null) {
+            return null;
+        }
+
+        User user = userCache.get(email);
+        if (user != null) {
+            return user;
+        }
+
+        final String username = userJson.optString("username", null);
+        if (username == null) {
+            return null;
+        }
+
+        user = new User(email, username);
+
+        user.name = userJson.optString("name", null);
+
+        userCache.put(email, user);
+
+        return user;
+    }
+
+    /**
+     * Parses a datetime string into a {@link Calendar} object. Will return null if the date cannot
+     * be parsed. Null-safe (will return null).
+     *
+     * @param datetime
+     *      the datetime to parse
+     * @return
+     *      the parsed {@link Calendar}, or null if it cannot be parsed.
+     */
+    private Calendar parseDatetime(@Nullable final String datetime) {
+        if (datetime == null) {
+            return null;
+        }
+
+        final Date parsedDate = DateUtils.parseDate(datetime, ISO8601);
+        if (parsedDate == null) {
+            return null;
+        }
+
+        final Calendar calendar = Calendar.getInstance();
+
+        calendar.clear();
+        calendar.setTimeZone(UTC);
```

Comment:
It was in the data I pulled from the server.  FWIW, the datetime parsing doesn't accept the timezone offset since java can't deal with "Z" (and, I'd need to do a bunch of string replacement to work around that).  Java fixes their timezone offset stuff in java 8 to work with zulu iso8601.

It doesn't look like lovemonster forces UTC either (so, it'll be server timezone):
https://github.groupondev.com/prod-tools/love/blob/master/app/models/love.rb#L42

Probably worth making the app parsing code more defensive about it.